### PR TITLE
[Modelzoo] DBMTL model.

### PR DESCRIPTION
**Pull request by jmaksymc**

### CODE REVIEW

**Comment by Duyi-Wang** [comment]
acc calculation in line 209 need update.`self.acc, self.acc_op = tf.metrics.accuracy(labels=self.label, predictions=self.model)`
'tf.metrics.accuracy'  don't accept probability as predictions. 
Please update to ``self.acc, self.acc_op = tf.metrics.accuracy(labels=self.label, predictions=tf.round(self.model))``

File: modelzoo/DBMTL/README.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    - `--checkpoint`: Full path to checkpoints input/output directory, default is `$(OUTPUT_DIR)/model_$(MODEL_NAME)_$(TIMESTAMPS)`
+    - `--steps`: Set the number of steps on train dataset. Default will be set to 10 epoch.
+    - `--batch_size`: Batch size to train. Default is 512.
+    - `--timeline`: Save steps of profile hooks to record timeline, zero to close, defualt to 0.
+    - `--save_steps`: Set the number of steps on saving checkpoints, zero to close. Default will be set to 0.
+    - `--keep_checkpoint_max`: Maximum number of recent checkpoint to keep. Default is 1.
+    - `--learning_rate`: Learning rate for network. Default is 0.1.
+    - `--bf16`: Enable DeepRec BF16 feature in DeepRec. Use FP32 by default.
+    - `--no_eval`: Do not evaluate trained model by eval dataset.
+    - `--protocol`: Set the protocol("grpc", "grpc++", "star_server") used when starting server in distributed training. Default to grpc.
```
Comment (aalbersk) [inline modelzoo/DBMTL/README.md:87]: Add `inter`, `intra`, `input_layer_partitioner` and `dense_layer_partitioner` information to README

File: modelzoo/DBMTL/README.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      - [Test Environment](#test-environment)
+      - [Performance Result](#performance-result)
+  - [Dataset](#dataset)
+    - [Prepare](#prepare)
+    - [Fields](#fields)
+    - [Processing](#processing)
+  - [TODO LIST](#todo-list)
+
+## Model Structure
+[Deep Bayesian Multi-Target Learning]DBMLT.
```
Comment (aalbersk) [inline modelzoo/DBMTL/README.md:19]: Could you add url to the paper on DBMLT?

File: modelzoo/DBMTL/train.py [hunk 3]
```diff
@@ -0,0 +1,709 @@
...
+import tensorflow as tf
+import os
+import sys
+import math
+import collections
+from tensorflow.python.client import timeline
+import json
+from tensorflow.python.ops import partitioned_variables
+
+if tf.__version__ >= '2.0':
```
Comment (aalbersk) [inline modelzoo/DBMTL/train.py:13]: Have you checked whether the script works on tf2? It might not due to `tf.data.Iterator`.

File: modelzoo/DBMTL/train.py [hunk 4]
```diff
@@ -0,0 +1,709 @@
...
+        features = all_columns
+        return features, label
+
+    # Extract lines from input files using the Dataset API.
+    dataset = tf.data.TextLineDataset(filename)
+    dataset = dataset.shuffle(buffer_size=400000,
+                              seed=SEED)  # set seed for reproducing
+    dataset = dataset.repeat(num_epochs)
+    dataset = dataset.batch(batch_size)
+    dataset = dataset.map(parse_csv, num_parallel_calls=28)
```
Comment (aalbersk) [inline modelzoo/DBMTL/train.py:98]: I guess it might be better to change `28` to `tf.data.experimental.AUTOTUNE` but I'm not sure it will change the performance.

File: modelzoo/DBMTL/train.py [hunk 5]
```diff
@@ -0,0 +1,709 @@
...
+        for key in TAG_COLUMN:
+            self.feature[key] = tf.strings.split(self.feature[key], '|')
+
+        with tf.variable_scope('input_layer',
+                               partitioner=self.__input_layer_partitioner,
+                               reuse=tf.AUTO_REUSE):
+            input_emb = tf.feature_column.input_layer(self.feature,
+                                                  self.__feature_column)
+        # Shared Layer
+        with tf.variable_scope(f'bottom_dnn_'):
```
Comment (aalbersk) [inline modelzoo/DBMTL/train.py:241]: fyi, You don't need to use f-string formatting if you are not passing any variable to the string.

File: modelzoo/DBMTL/README.md (hunk 1 above)
Comment (jmaksymc) [reply modelzoo/DBMTL/README.md]: Changed

File: modelzoo/DBMTL/train.py (hunk 3 above)
Comment (jmaksymc) [reply modelzoo/DBMTL/train.py]: Removed

File: modelzoo/DBMTL/train.py (hunk 5 above)
Comment (jmaksymc) [reply modelzoo/DBMTL/train.py]: Changed

**Comment by jmaksymc** [comment]
Update on jira [WC1-1043]

File: modelzoo/DBMTL/train.py [hunk 6]
```diff
@@ -0,0 +1,681 @@
...
+                                                'DBMTL_checkpoint'),
+                                                global_step=_in)
+                    print(f'Save checkpoint to {checkpoint_path}')
+                elif (args.timeline > 0 and _in % args.timeline == 0):
+                    _, train_loss = sess.run([model.train_op, model.loss],
+                                            options=options,
+                                            run_metadata=run_metadata)
+                    fetched_timeline = timeline.Timeline(run_metadata.step_stats)
+                    chrome_trace = fetched_timeline.generate_chrome_trace_format()
+                    print(f'Save timeline to {checkpoint_path}')
```
Comment (Duyi-Wang) [inline modelzoo/DBMTL/train.py:397]: `checkpoint_path` to `checkpoint_dir`

Comment (jmaksymc) [reply modelzoo/DBMTL/train.py]: Changed `checkpoint_path` to `checkpoint_dir`

**Comment by jmaksymc** [comment]
Changed `SEED` to `args.seed`.
